workers/data_refinery_workers/downloaders/array_express.py

```diff
@@ -23,44 +75,29 @@ def download_array_express(job_id):
     success = True
     utils.start_job(job)
 
-    try:
-        batch = Batch.objects.get(id=job.batch_id)
-    except ObjectDoesNotExist:
-        logger.error("Cannot find batch record with ID %d.", job.batch_id)
-        utils.end_job(job, None, False)
-        return
+    batch_relations = DownloaderJobsToBatches.objects.filter(downloader_job_id=job_id)
+    batches = list(map(lambda x: x.batch, batch_relations))
```

Comment:
Oh yes, that's very nice. List comprehensions are probably the feature of python I need to become comfortable with the most.